apis/gateway/v1beta1/apirule_types.go

```diff
@@ -54,7 +54,6 @@ type APIRuleSpec struct {
 	// +optional
 	CorsPolicy *CorsPolicy `json:"corsPolicy,omitempty"`
 	// Represents the array of Oathkeeper access rules to be applied.
-	// +kubebuilder:validation:MinItems=1
```

Comment:
When I spoke with @barchw, he mentioned if I remember correctly, that it's related to the conversion between version of the APIRule.

Reply:
If I understand it correctly the reason might be ExtAuth. I didn't review the whole PR, but I understand the following issue. When v2alpha1 is configured with only ExtAuth there will be no rule in the v1beta1 version, because the ExtAuth configuration is only stored as a annotation on the v1beta1 version.
To workaround this, the min items was removed. But unfortunately, with this we allow the creation of v1beta1 rules without any rule. I don't know why this decision was made, but I find it confusing.
The other solution I can see is to also support ExtAuth in v1beta1 handlers.

Reply:
That's correct

Reply:
I will do it with cel validation (if v1beta1 validate, otherwise don't)

Reply:
Resolved by storing simplified ext-auth https://github.com/kyma-project/api-gateway/pull/1258/commits/bd059ba43d8bbae3c411eb6ce58c404d7c7920e4